tests/unit/test_fetch_configs.py

```diff
@@ -246,6 +246,34 @@ def test_unknown_os(self):
             get_build_regex('test', 'unknown', 32, 'x86')
 
 
+class TestFallbacksConfig(TestFirefoxConfigLinux64):
+    def setUp(self):
+        self.conf = create_config(self.app_name, self.os, self.bits,
+                                  self.processor)
+        self.conf.BUILD_TYPE_FALLBACKS = {
+            'opt': ['test', 'fallback']
+        }
+        self.conf.set_build_type('opt')
+
+    def test_fallbacking(self):
+        assert self.conf.build_type == 'opt'
+        self.conf.inc_used_build()
+        assert self.conf.build_type == 'test'
+        self.conf.inc_used_build()
+        assert self.conf.build_type == 'fallback'
+        # Check we wrap
+        self.conf.inc_used_build()
+        assert self.conf.build_type == 'opt'
+
+    def test_fallback_routes(self):
+        routes = list(self.conf.tk_inbound_routes(
+            create_push('1a', TIMESTAMP_GECKO_V2)
+        ))
+        assert len(routes) == 3
+        assert routes[0].endswith('-opt')
+        assert routes[2].endswith('-fallback')
```

Comment:
Could you assert on the specific return values here?

Reply:
Done.